### ISSUE TITLE
Nomad Provider support for Unix Domain Socket Task API

### ISSUE THREAD

**Issue by mhehle** [traefik/traefik#10044]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you expect to see?

currently the nomad provider configuration requires an http(s) address as endpoint. 

``` ini
[providers.nomad]
  exposedByDefault = false
[providers.nomad.endpoint]
  address = "https://127.0.0.1:4646"
  token = "acl-token-goes-in-here"
  [providers.nomad.endpoint.tls]
    ca = "/path/to/ca.pem"
    cert = "/path/to/cert.pem"
    key = "/path/to/key.pem"
```

Nomad also provides a Task API, that is exposed to every job.

**Taken from the Nomad documentation:**
> Nomad's Task API provides every task managed by Nomad with a Unix Domain Socket (UDS) to access the local agent's > HTTP API. Regardless of agent configuration the Task API does not require mTLS, but always requires authentication. 
> The Unix Domain Socket is located at ${NOMAD_SECRETS_DIR}/api.sock.

### Example

``` ini
[providers.nomad]
  exposedByDefault = false
[providers.nomad.endpoint]
  address = "unix:///.../secrets/api.sock"
  token = "acl-token-goes-in-here"
```

### Benefits
- simple configuration
- no additional mTLS certificate generation and maintenance

**Comment by SamMousa** [comment]
I don't think this is a traefik issue, it is an issue with the nomad API client that they import.

**Comment by mhehle** [comment]
I wouldn't call that a traefik problem either, more of an improvement.

**Comment by SamMousa** [comment]
Given that they will shortly support this in Nomad via https://github.com/hashicorp/nomad/pull/16872, can we get some eyes in this from Traefik? 

I reckon all you'd need to do is update the nomad library, and of course go through your approval / QA workflows.

**Comment by IamTheFij** [comment]
This is working now that the Nomad client has been updated.

**Comment by mhehle** [comment]
nice! I have just tested the feature on v3.0.1 and it works.

**Example Nomad Template:**
```yaml
template {
        data        = <<EOH
{{ with nomadVar "nomad/jobs/traefik" }}
providers:
  nomad:
    endpoint:
      address: 'unix://{{ env "NOMAD_SECRETS_DIR" }}/api.sock'
      token: '{{ .traefik_acl_token }}'       
{{ end }}
EOH
        destination = "local/traefik.yml"
}
```

**Comment by SamMousa** [comment]
Have you tried using the workload identity token as well?

https://developer.hashicorp.com/nomad/docs/job-specification/identity

I'm excited to deploy this to our infra!

**Comment by mhehle** [comment]
@SamMousa yes it works with workload identity too.

Example Nomad Template with workload identity:
```hcl
template {
        data        = <<EOH
{{ with nomadVar "nomad/jobs/traefik" }}
providers:
  nomad:
    endpoint:
      address: 'unix://{{ env "NOMAD_SECRETS_DIR" }}/api.sock'
      token: '{{ env "NOMAD_TOKEN" }}'
{{ end }}
EOH
        destination = "local/traefik.yml"
}
```
your job needs to expose the workload identity as ENV VAR

```hcl
job "traefik" {
  type        = "service"
  group "default" {
    task "main" {
      identity {
        env = true
      }
    }
  }
}
```

you also need to apply an acl policy for the traefik job

**Policy**
```hcl
namespace "default" {
  policy = "read"
  capabilities = ["list-jobs","read-job"]
}
```
`nomad acl policy apply -namespace default -job traefik traefik-policy ./policy.hcl`

**Comment by Jamesits** [comment]
Please note that workload identity exposed as an env var do expire after a few days.

**Comment by IamTheFij** [comment]
If deployed with Nomad, the template will re-render when it expires and changed.

**Comment by dannyhpy** [comment]
> If deployed with Nomad, the template will re-render when it expires and changed.

If it is exposed through an environment variable though, it won't change the environment AFAIK.
You need to append `"change_mode": "restart"` to the `identity` block as a workaround to restart the task once the token is expired.
